lightning/src/ln/channel.rs

```diff
@@ -387,6 +387,52 @@ impl fmt::Display for ChannelError {
 	}
 }
 
+pub(super) struct WithChannelContext<'a, 'b, L: Deref, Signer: ChannelSigner> where L::Target: Logger {
```

Comment:
There are a lot of places where this is created multiple times in a single function. Given this doesn't copy anything, can't we wrap it only once in those cases where we aren't calling a `&mut self` method?

Reply:
will do